packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -31,21 +29,12 @@ const zSessionStopEventType = '_session.stop';
 class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
   /// {@macro amplify_analytics_pinpoint_dart.amplify_analytics_pinpoint_dart}
   AmplifyAnalyticsPinpointDart({
-    SecureStorageInterface? endpointInfoStore,
     CachedEventsPathProvider? pathProvider,
     AppLifecycleProvider? appLifecycleProvider,
-    DeviceContextInfoProvider? deviceContextInfoProvider,
-    LegacyNativeDataProvider? legacyNativeDataProvider,
-  })  : _endpointInfoStore = endpointInfoStore ??
-            AmplifySecureStorageWorker(
-              config: AmplifySecureStorageConfig(
-                scope: 'analyticsPinpoint',
-              ),
-            ),
+    AnalyticsClient? analyticsClient,
```

Comment:
Since the AnalyticsClient will handle the endpointInfoStore, deviceContextInfo, legacyNativeData etc. parameters, they can now be removed from plugin class.